adding g1gc and gc log/detail attributes

msilvey opened the pull request:
I would like to be able to use this chef recipe with the G1 GC enabled and/or be able to enable GC logging for all my Cassandra nodes via the template.

michaelklishin commented on templates/default/cassandra-env.sh.erb:
```diff
@@ -273,6 +282,19 @@ if [ "$JVM_VERSION_SHORT" = "1.7" ] && [ "$JVM_ARCH" = "64-Bit" ] ; then
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCApplicationStoppedTime"
+JVM_OPTS="$JVM_OPTS -XX:+PrintPromotionFailure"
+JVM_OPTS="$JVM_OPTS -XX:PrintFLSStatistics=1"
+JVM_OPTS="$JVM_OPTS -Xloggc:/var/log/cassandra/gc.log"
```
I think this should respect Cassandra directory attributes.

michaelklishin commented on templates/default/cassandra-env.sh.erb:
```diff
@@ -258,14 +258,23 @@ fi
 JVM_OPTS="$JVM_OPTS -XX:+UseParNewGC"
 JVM_OPTS="$JVM_OPTS -XX:+UseConcMarkSweepGC"
 JVM_OPTS="$JVM_OPTS -XX:+CMSParallelRemarkEnabled"
-JVM_OPTS="$JVM_OPTS -XX:SurvivorRatio=<%= node['cassandra']['gc_survivor_ratio'] %>"
```
Why were these attributes removed in favour of hardcoded values?

msilvey commented:
I updated for log_dir and added the configurable options back properly.  I did some cut and pasting from the older version we are using and didn't check things properly.

michaelklishin commented:
@msilvey thank you!
